Makefile

```diff
@@ -0,0 +1,33 @@
+DOCKER_IMAGE = oam/server:latest
+
+all: server
+
+server:
+	@docker build -f ./server/Dockerfile -t $(DOCKER_IMAGE) ./server
+
+start: server
+	@docker run \
+		--detach \
+		--name oam-server \
+		--publish 8000:8000 \
+		--volume $(PWD)/server:/app \
+		$(DOCKER_IMAGE) start
+
+test: start
+	@sleep 1
+
+	@docker run \
+		--rm \
+		--name oam-server-test \
+		--link oam-server:oam-server \
+		--volume $(PWD)/server:/app \
```

Comment:
You left this comment on the `--volume` line, but referenced links. Going to try and explain both flags just to cover all of the bases:

- The `--link` there is to connect the `oam-server` container with the `oam-server-test` container. That link adds an entry to `/etc/hosts` of the `oam-server-test` container allowing the string `oam-server` resolve to the `oam-server` container's IP address.

- The `--volume` line mounts the local `/server` directory into `/app` inside of the container. We are doing a `COPY` during the image build process, but we're using `--volume` here to catch any code changes that may have happened between building the image and now. The volume will end up clobbering the work `COPY` did (which is actually part of why the NPM symlink hack exists).

For example, you start tweaking an existing function, but want to run the test suite to make sure it still passes. This allows you to bypass rebuilding the image to test *every* time there is a code change.